lib/SqlFormatter.php

```diff
@@ -214,38 +221,33 @@ protected static function getNextToken($string, $previous = null)
         }
 
         // Comment
-        if ($string[0] === '#' || (isset($string[1])&&($string[0]==='-'&&$string[1]==='-') || ($string[0]==='/'&&isset($string[1])&&$string[1]==='*'))) {
-            // Comment until end of line
-            if ($string[0] === '-' || $string[0] === '#') {
-                $last = strpos($string, "\n");
-                $type = self::TOKEN_TYPE_COMMENT;
-            } else { // Comment until closing comment tag or end of query
-                if (strpos($string, "*/", 2) !== false) {
-                    $last = strpos($string, "*/", 2) + 2;
-                } else {
-                    $last = strlen($string);
-                }
-                $type = self::TOKEN_TYPE_BLOCK_COMMENT;
-            }
-
-            if ($last === false) {
-                $last = strlen($string);
+        $found = false;
+        foreach (self::$comment_tokens as $comment) {
+            $start = $comment[0];
+            $end = isset($comment[1]) ? $comment[1] : "\n";
+            $found = substr($string, 0, strlen($start)) === $start;
+            if ($found) {
+                break;
             }
-
+        }
+        if ($found) {
+            /** @var string $start */
+            /** @var string $end */
+            $endPos = strpos($string, $end, strlen($start));
+            $last = $endPos ?
+                ($endPos + ($end === "\n" ? 0 : strlen($end))) : strlen($string);
```

Comment:
🤯

Reply:
Hej, mas pravdu :) 

Spravil som to viac citatelnejsie:
https://github.com/keboola/sql-formatter/pull/3/commits/94fee9b147c0cf7f3f57c640c01c6a88bed541ff

Reply:
jj tak to je lepší 👍  díky